2-Draft-Accepted/RFCxxxx-PowerShellGet-3.0.md

```diff
@@ -0,0 +1,476 @@
+a prompt will appear _before anything is installed_, listing all the resources
+including module name, version, size, and repository to be installed unless
+`-IncludeDependencies` is specified which will install without prompting.
+Rejecting the prompt will result in nothing being installed.
+
+Declared dependencies are searched and installed using the same trust algorithm
+as described for `Install-PSResource` above.
+
+>[!NOTE]
+>Installing dependencies is following the `apt` experience in that it prompts
+>by default instead of automatically installing dependencies.  Since dependencies
+>can be quite large, this would be equivalent to `ConfirmImpact=High` which
+>would prompt by default.
+>Dependency installation works with `-DestinationPath` parameter.
+
+We will also introduce a `New-RequiredResourceFile` cmdlet which will create a template file.
+If the switch `-AsPSD1` is used it will create a psd1 file, otherwise it will default to json. 
+
+### Saving resources
+
+This cmdlet uses `Install-PSResource -DestinationPath`.
+
+With the removal of PackageManagement, there is still a need to support saving
+arbitrary nupkgs (assemblies) used for scripts.
+
+`Save-PSResource -Type Library` will download nupkgs that have a `lib` folder.
+The dependent native library in `runtimes` matching the current system runtime
+will be copied to the root of the destination specified.
+A `-IncludeAllRuntimes` can be used to explicitly retain the `runtimes` directory
+hierarchy within the nupkg to the root of the destination.
+
+>[!NOTE]
+>Library support may not be available in 3.0.
+A `-Prerelease` switch allows saving prerelease versions.
+If the `-Version` includes a prerelease label like `2.0.0-beta4`, then this
+switch is not necessary and the prerelease version will be installed.
+
+A `-AsNupkg` switch will save the resource as a nupkg (if it was originally a
+nupkg) instead of expanding it into a folder.
+
+### Updating resources
+
+`Update-PSResource` will update all resources to most recent minor version by default.
+
+A `-UpdateTo` parameter has values `MinorVersion` (as default), `MajorVersion`, `PatchVersion`.
+`MajorVersion` allows updating to latest major version.
+`PatchVersion` allows updating to latest patch version (e.g. 6.2.1 to 6.2.4).
+
+If the installed resource is a pre-release, it will automatically update to
+latest prerelease or stable version (if available).
+
+### Publishing resources
+
+`Publish-PSResource` will supporting publishing modules, scripts and nupkgs.
+It will follow the same model as `Publish-Module` in v2.
+
+A `-DestinationPath` can be used to publish the resulting nupkg locally.
+`Publish-PSResource` can accept a path to the nupkg to publish to a repository.
+
+A `-Nuspec` parameter can be used to specify a nuspec file rather than relying
+on this module to produce one.
+
+A `-SkipDependenciesCheck` switch can be used to bypass the default check that
+all dependencies are present.

```

Comment:
... all dependencies are present where? on the local system or in the target repository?